api/src/main/java/org/openmrs/ObsReferenceRange.java

```diff
@@ -0,0 +1,264 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import org.hibernate.search.annotations.DocumentId;
+import org.hibernate.search.annotations.Field;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+import java.util.Date;
+
+/**
+ * ObsReferenceRange is typically a reference range of an Observation 
+ * The reference range is created at the point of recording {@link Obs}
+ *
+ * @since 2.7.0
+ */
+@Entity
+@Table(name = "obs_reference_range")
+public class ObsReferenceRange extends BaseReferenceRange implements Creatable, Voidable, java.io.Serializable {
```

Comment:
I do not think we need the obs reference range to be voided or created separately from its parent obs. So i guess we can get rid of these interfaces. Do you have cases when you feel otherwise?

Reply:
Yeah, the more I was thinking about it, the more I was wondering if it made sense to turn this into an `@Embeddable` class, so that, effectively, there's just the Obs with the reference range as basically part of it?

Reply:
So we make these ranges part of Obs entity (and make this class Embeddable) right? To me it makes sense.

Reply:
Update: @ibacher  we had a discussion with @dkayiwa  and we saw it well to have `ObsReferencRange` with it's own table instead of adding all these new columns in Obs.